File: 28-implement-strstr/28-implement-strstr.c
int strStr(char * haystack, char * needle){
    int i,j;
    int m,n;
    if (needle == "")
        return 0;
    m = strlen(haystack);
    n = strlen(needle);
    if (m < n)
        return -1;
    
    for(i = 0; i <= m-n; ++i)
    {
        for(j = 0; j < n; ++j)
        {
            if(haystack[i+j] != needle[j])
                break;
        }
        if (j == n)
            return i;
    }
    return -1;
}

File: 28-implement-strstr/28-implement-strstr.java
class Solution {
    public int strStr(String haystack, String needle) {
    int m = haystack.length();
    int n = needle.length();
    
    if(n == 0)
        return 0;
    if(n > m)
        return -1;
    
    for(int i = 0; i < m-n+1; i++){
        if(haystack.substring(i,i+n).equals(needle)){
            return i;
        }
    }
    return -1;
}
}